third_party/maya/lib/usdMaya/usdWriteJob.cpp

```diff
@@ -348,11 +348,13 @@ void usdWriteJob::endJob()
         // prim for the export... usdVariantRootPrimPath
         mJobCtx.mStage->GetRootLayer()->SetDefaultPrim(defaultPrim);
     }
+    // clear this so that no stage references are left around
+    // and running code in destructors
+    mJobCtx.mMayaPrimWriterList.clear();
```

Comment:
I think having an explicit hook would be preferred to doing stuff in a destructor here - I'd rather have an explicit call to some sort of "PostExport" callback (or virtual function on the MayaPrimWriter base class)